core/src/main/java/io/kestra/core/test/TestSuiteRunResult.java

```diff
@@ -0,0 +1,24 @@
+package io.kestra.core.test;
+
+import io.kestra.core.test.flow.UnitTestResult;
+
+import java.util.List;
+
+
+public record TestSuiteRunResult(
+    String id,
+    String testSuiteId,
+    Status status,
+    List<UnitTestResult> results) {
+
+    public static TestSuiteRunResult of(String id, String testSuiteId, List<UnitTestResult> results) {
+        var status = results.stream().anyMatch(result -> result.status().equals(UnitTestResult.Status.FAILED)) ? Status.FAILED : Status.SUCCESS;
+        return new TestSuiteRunResult(id, testSuiteId, status, results);
+    }
+
+    public enum Status {
```

Comment:
```suggestion
    public enum State {
```

We use state everywhere.

I also wonder if we didn't need an ERROR state to make the distinction between assertion failures and assertion errors. Like in JUnit.